Assets/Scripts/Environment/BreakableWall.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BreakableWall : MonoBehaviour
+{
+    /* Notes:
+     * The wall object with this script must also have a collider component
+     */
+    public int wallHealth = 10;
+    public int playerCollisionDamage= 5;
+
+    // Update is called once per frame
+    void Update()
+    {
+        if (wallHealth <= 0)
+        {
+            DestroyWall();
+        }
+
+    }
+
+    void DestroyWall()
+    {
+        //Debug.Log("Wall Destroyed");
+        Destroy(gameObject);
+    }
+    public void OnTriggerEnter(Collider other)
+    {
+        if (other.gameObject.CompareTag("Player"))
+        {
+            //Debug.Log("Hit player");
+            //collision.gameObject.GetComponent<PlayerHealth>().TakeDamage(playerCollisionDamage);
+
+            DestroyWall();
```

Comment:
There should be another method in the script that applies the damage to the wall, this method should take an int as a parameter(the damage being applied), in that method you should decrease the health, and after decreasing it check if the health is <= 0 and if it is call the destroyWall method is called